src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -76,7 +76,7 @@ public static NullableFlowState GetNullableState(ArrayBuilder<TypeWithState> typ
                     if (type.ContainsErrorType())
                     {
                         inferredFromFunctionType = false;
-                        return type;
+                        return type is FunctionTypeSymbol function ? function.GetInternalDelegateType() : type;
```

Comment:
>function.GetInternalDelegateType() [](http://example.com/codeflow?start=69&length=34)

Should we set `inferredFromFunctionType = true` in this case?

Reply:
> Should we set `inferredFromFunctionType = true` in this case?

It looks to me that we didn't really infer anything, but grabbed the first type with errors, there could be other type candidates that aren't function types. Since this is already an error scenario, I do not think adjusting the value is going to have any positive effect. However, I am willing to rely on your expertise in this area and make adjustments based on your recommendations. Do you have any specific scenario, that can benefit from the change, in mind?

Reply:
> No, I don't have a specific scenario in mind that would benefit from the change. But since the returned type was inferred from a function type, it may be surprising to a caller that does depend on that behavior in the future.

Could you please clarify, are you making a specific suggestion or not?